Book_Summarizer/data_download_and_stats.py

```diff
@@ -71,74 +72,74 @@ def calculate_author_match(author1,author2):
-  try:
-    wget.download(web_page)
-  except:
-    print("404 error")
-    file_exists = False
-  return file_exists
+    web_page = "http://aleph.gutenberg.org/"
+    for d in str(pg_id)[:-1]:
+        web_page = web_page + str(d) + "/"
+    web_page = web_page + str(pg_id) + "/" + str(pg_id) + ".zip"
+    file_exists = True
+    try:
+        wget.download(web_page)
+    except:
+        print("404 error")
+        file_exists = False
+    return file_exists
 
 
 # extract_book(pg_index)
 #
 # unzip the book and move to books folder
 # remove zip file afterwards
 def extract_book(pg_index, zip_filename, text_filename, book_filename):
-  with ZipFile(zip_filename, 'r') as zipObj:
-     zipObj.extractall()
-  if os.path.exists(text_filename):
-    shutil.move(text_filename,book_filename)
-  else:
-    # some files have an extra folder before the file
-    shutil.move(str(pg_index) + '/' + text_filename, book_filename)
-    shutil.rmtree(str(pg_index))
-  os.remove(zip_filename)
+    with ZipFile(zip_filename, 'r') as zipObj:
+        zipObj.extractall()
+    if os.path.exists(text_filename):
+        shutil.move(text_filename, book_filename)
+    else:
+        # some files have an extra folder before the file
+        shutil.move(str(pg_index) + '/' + text_filename, book_filename)
+        shutil.rmtree(str(pg_index))
+    os.remove(zip_filename)
 
 
 # save_summary(df_summaries, new_title, summary_filename)
 #
 # saves the summary to the summary folder
 def save_summary(df_summaries, new_title, summary_filename):
-  new_summary = df_summaries[df_summaries[2]==new_title][6].to_string()[6:]
-  with open(summary_filename, 'w') as f:
-    f.write(new_summary)
+    new_summary = df_summaries[df_summaries[2] == new_title][6].to_string()[6:]
+    with open(summary_filename, 'w') as f:
+        f.write(new_summary)
 
 
 # save_clean_book(book_filename,clean_book_filename)
 #
 # removes information about project gutenberg from the book
 # replaces the book file with the cleaned book
-def save_clean_book(book_filename,clean_book_filename):
-  book = open(book_filename,'r', encoding='latin-1')
-  clean_book = open(clean_book_filename,'w')
-  write_lines = False
-  # TODO: some of the older books do not match this formatting
-  # and end up with no lines of text output
-  for l in book:
-    if (l[:12]=='*** START OF') or (l[:11]=='***START OF') or (l[:11]=='*END*THE SM'):
-      write_lines = True
-    elif (l[:10]=='*** END OF') or (l[:9]=='***END OF'):
-      write_lines = False
-    elif write_lines:
-      clean_book.write(l)
-  book.close()
-  clean_book.close()
-  # if the formatting didn't match the above, just use the complete
-  # book with project gutenberg information
-  if os.stat(clean_book_filename).st_size == 0:
-    book = open(book_filename,'r', encoding='latin-1')
-    clean_book = open(clean_book_filename,'w')
+def save_clean_book(book_filename, clean_book_filename):
+    book = open(book_filename, 'r', encoding='latin-1')
```

Comment:
Actually, am curious why `latin-1` instead of something like `utf-8`, as is more common. Are all the texts encoded in `latin-1`?

Reply:
Most are utf-8. Using latin-1 encoding prevented errors for those texts that are not utf-8.